### PR TITLE
nv-codec-headers: 12.0.16.0 -> 12.1.14.0, ffmpeg: 6.0 -> 6.1, jellyfin-ffmpeg: 6.0-7 -> 6.0.1-1

### PR DESCRIPTION
**Pull request by arthsmn**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Atemu — CHANGES_REQUESTED** [review]
Please rebase away the merge commit.

**Comment by arthsmn** [comment]
> Please rebase away the merge commit.

Done.

File: pkgs/development/libraries/ffmpeg/generic.nix
Comment (chayleaf) [inline pkgs/development/libraries/ffmpeg/generic.nix:442]:
```suggestion
    (optionalString withAribcaption "--enable-libaribcaption")
```

**Review by chayleaf — CHANGES_REQUESTED** [review]

**Review by chayleaf — APPROVED** [review]
might be worth pinging gifsik and wl-screenrec maintainers after merging so they know about the switch to ffmpeg_5 (it would be hard to notice the change in all-packages.nix)

**Comment by arthsmn** [comment]
> might be worth pinging gifsik and wl-screenrec maintainers after merging so they know about the switch to ffmpeg_5 (it would be hard to notice the change in all-packages.nix)

Done.

File: pkgs/development/libraries/ffmpeg/generic.nix
Comment (marsam) [inline pkgs/development/libraries/ffmpeg/generic.nix:442]: Shouldn't this be `enableFeature`?
```suggestion
    (enableFeature withAribcaption "--enable-libaribcaption")
```

File: pkgs/servers/mpd/default.nix
Comment (marsam) [inline pkgs/servers/mpd/default.nix:153]: I think we can use `fetchpatch` with https://github.com/MusicPlayerDaemon/MPD/commit/50c1e3738a672f933560247b5527d365d17b1c02

**Review by ghost — APPROVED** [review]

**Review by Atemu — APPROVED** [review]
Diff LGTM.

Do you not want to add yourself as a maintainer anymore?

**Review by chayleaf — APPROVED** [review]
there's a couple of comment typos but i'm not gonna nitpick comments

File: pkgs/development/libraries/ffmpeg/generic.nix
Comment (arthsmn) [reply pkgs/development/libraries/ffmpeg/generic.nix]: Done.

**Comment by Atemu** [comment]
> I think it's better to get more familiar with Nix and the Nixpkgs workflow before maintaining such an important package. I'll just do PRs for now.

Nah, you're fine. Maintainership "only" means that people come to you if the package needs to change. You'll get pinged on any PRs that touches ffmpeg for example.

What you need to maintain a package is knowledge about the package's specifics; how to test it still works etc. General Nix/Nixpkgs experience always helps of course but that's not the important part here.

**Comment by arthsmn** [comment]
> > I think it's better to get more familiar with Nix and the Nixpkgs workflow before maintaining such an important package. I'll just do PRs for now.
> 
> Nah, you're fine. Maintainership "only" means that people come to you if the package needs to change. You'll get pinged on any PRs that touches ffmpeg for example.
> 
> What you need to maintain a package is knowledge about the package's specifics; how to test it still works etc. General Nix/Nixpkgs experience always helps of course but that's not the important part here.

Oh thanks for the clarification, I think I'll add myself so.

**Comment by Atemu** [comment]
@ofBorg eval

**Comment by Atemu** [comment]
Result of `nixpkgs-review pr 271905` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>59 packages built:</summary>
  <ul>
    <li>ffmpeg</li>
    <li>ffmpeg-headless</li>
    <li>ffmpeg-headless.bin</li>
    <li>ffmpeg-headless.data</li>
    <li>ffmpeg-headless.dev</li>
    <li>ffmpeg-headless.doc</li>
    <li>ffmpeg-headless.lib</li>
    <li>ffmpeg-headless.man</li>
    <li>ffmpeg-normalize</li>
    <li>ffmpeg-normalize.dist</li>
    <li>ffmpeg.bin</li>
    <li>ffmpeg.data</li>
    <li>ffmpeg.dev</li>
    <li>ffmpeg.doc</li>
    <li>ffmpeg.lib</li>
    <li>ffmpeg.man</li>
    <li>ffmpeg_4</li>
    <li>ffmpeg_4-full</li>
    <li>ffmpeg_4-full.bin</li>
    <li>ffmpeg_4-full.data</li>
    <li>ffmpeg_4-full.dev</li>
    <li>ffmpeg_4-full.doc</li>
    <li>ffmpeg_4-full.lib</li>
    <li>ffmpeg_4-full.man</li>
    <li>ffmpeg_4-headless</li>
    <li>ffmpeg_4-headless.bin</li>
    <li>ffmpeg_4-headless.data</li>
    <li>ffmpeg_4-headless.dev</li>
    <li>ffmpeg_4-headless.doc</li>
    <li>ffmpeg_4-headless.lib</li>
    <li>ffmpeg_4-headless.man</li>
    <li>ffmpeg_4.bin</li>
    <li>ffmpeg_4.data</li>
    <li>ffmpeg_4.dev</li>
    <li>ffmpeg_4.doc</li>
    <li>ffmpeg_4.lib</li>
    <li>ffmpeg_4.man</li>
    <li>ffmpeg_5</li>
    <li>ffmpeg_5-full</li>
    <li>ffmpeg_5-full.bin</li>
    <li>ffmpeg_5-full.data</li>
    <li>ffmpeg_5-full.dev</li>
    <li>ffmpeg_5-full.doc</li>
    <li>ffmpeg_5-full.lib</li>
    <li>ffmpeg_5-full.man</li>
    <li>ffmpeg_5-headless</li>
    <li>ffmpeg_5-headless.bin</li>
    <li>ffmpeg_5-headless.data</li>
    <li>ffmpeg_5-headless.dev</li>
    <li>ffmpeg_5-headless.doc</li>
    <li>ffmpeg_5-headless.lib</li>
    <li>ffmpeg_5-headless.man</li>
    <li>ffmpeg_5.bin</li>
    <li>ffmpeg_5.data</li>
    <li>ffmpeg_5.dev</li>
    <li>ffmpeg_5.doc</li>
    <li>ffmpeg_5.lib</li>
    <li>ffmpeg_5.man</li>
    <li>ffmpegthumbnailer</li>
  </ul>
</details>

**Comment by arthsmn** [comment]
@Atemu @chayleaf @justinas and all the others that helped here. I just wanted to say thank you! I've been having only great experiences in this community. It's been 2 weeks since I started on NixOS and I'm already making a contribution this big! Have a great day you all!

**Comment by arthsmn** [comment]
Also, I think #260538 can be closed.

**Comment by Atemu** [comment]
You're welcome @arthsmn :)

**Comment by OPNA2608** [comment]
Not familiar with the stability of these packages across minor version bumps, but is this backportable to 23.11? `punes` on current 23.11 segfaults on startup:

```
Thread 1 ".punes-wrapped" received signal SIGSEGV, Segmentation fault.
0x00007ffff5a50147 in av_fifo_read () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavutil.so.58
(gdb) bt
#0  0x00007ffff5a50147 in av_fifo_read () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavutil.so.58
#1  0x00007ffff5cac3c2 in reorder_queue_flush () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavcodec.so.60
#2  0x00007ffff5cae0d1 in ff_nvenc_encode_close () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavcodec.so.60
#3  0x00007ffff5c7ecca in avcodec_close () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavcodec.so.60
#4  0x00007ffff5c7ee34 in avcodec_open2[cold] () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavcodec.so.60
#5  0x0000000000567176 in recording_init () at /build/source/src/core/recording.c:206
#6  0x0000000000454c3d in main (argc=<optimized out>, argv=0x7fffffffc978) at /build/source/src/core/main.c:95
```

Cherry-picking this merge onto `nixos-23.11` fixes the crash locally.

**Comment by arthsmn** [comment]
> Not familiar with the stability of these packages across minor version bumps, but is this backportable to 23.11? `punes` on current 23.11 segfaults on startup:
> 
> ```
> Thread 1 ".punes-wrapped" received signal SIGSEGV, Segmentation fault.
> 0x00007ffff5a50147 in av_fifo_read () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavutil.so.58
> (gdb) bt
> #0  0x00007ffff5a50147 in av_fifo_read () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavutil.so.58
> #1  0x00007ffff5cac3c2 in reorder_queue_flush () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavcodec.so.60
> #2  0x00007ffff5cae0d1 in ff_nvenc_encode_close () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavcodec.so.60
> #3  0x00007ffff5c7ecca in avcodec_close () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavcodec.so.60
> #4  0x00007ffff5c7ee34 in avcodec_open2[cold] () from /nix/store/dkxlgdrafqv3r91w8mfm06zvp7m24xj5-ffmpeg-6.0-lib/lib/libavcodec.so.60
> #5  0x0000000000567176 in recording_init () at /build/source/src/core/recording.c:206
> #6  0x0000000000454c3d in main (argc=<optimized out>, argv=0x7fffffffc978) at /build/source/src/core/main.c:95
> ```
> 
> Cherry-picking this merge onto `nixos-23.11` fixes the crash locally.

I think they don't bump versions on stable releases, maybe @Atemu can give a way.

**Comment by ElvishJerricco** [comment]
This broke jellyfin-ffmpeg. It now fails with:

```
Unknown option "--enable-libaribcaption".
```

It can be fixed with

```diff
diff --git a/pkgs/development/libraries/jellyfin-ffmpeg/default.nix b/pkgs/development/libraries/jellyfin-ffmpeg/default.nix
index 7109a5d6dfd1..f519fb15b9b9 100644
--- a/pkgs/development/libraries/jellyfin-ffmpeg/default.nix
+++ b/pkgs/development/libraries/jellyfin-ffmpeg/default.nix
@@ -4,7 +4,7 @@
 , lib
 }:
 
-ffmpeg_6-full.overrideAttrs (old: rec {
+(ffmpeg_6-full.override { withAribcaption = false; }).overrideAttrs (old: rec {
   pname = "jellyfin-ffmpeg";
   version = "6.0.1-1";
```

But I have no idea if this is actually an acceptable fix. It looks like jellyfin-ffmpeg has not been updated with ffmpeg 6.1

**Comment by arthsmn** [comment]
> This broke jellyfin-ffmpeg. It now fails with:
> 
> ```
> Unknown option "--enable-libaribcaption".
> ```
> 
> It can be fixed with
> 
> ```diff
> diff --git a/pkgs/development/libraries/jellyfin-ffmpeg/default.nix b/pkgs/development/libraries/jellyfin-ffmpeg/default.nix
> index 7109a5d6dfd1..f519fb15b9b9 100644
> --- a/pkgs/development/libraries/jellyfin-ffmpeg/default.nix
> +++ b/pkgs/development/libraries/jellyfin-ffmpeg/default.nix
> @@ -4,7 +4,7 @@
>  , lib
>  }:
>  
> -ffmpeg_6-full.overrideAttrs (old: rec {
> +(ffmpeg_6-full.override { withAribcaption = false; }).overrideAttrs (old: rec {
>    pname = "jellyfin-ffmpeg";
>    version = "6.0.1-1";
> ```
> 
> But I have no idea if this is actually an acceptable fix. It looks like jellyfin-ffmpeg has not been updated with ffmpeg 6.1

First of all, jellyfin-ffmpeg will just be updated in version 7 of ffmpeg. This is a known issue and it's being discussed [here](https://github.com/NixOS/nixpkgs/issues/280645) and in the PRs associated with the issue.

**Pull request merged**
